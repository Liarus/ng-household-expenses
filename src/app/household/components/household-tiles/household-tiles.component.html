<div fxLayout="column">
  <div class="app-spinner-loading-shade" *ngIf="isLoading">
    <mat-spinner></mat-spinner>
  </div>
  <div fxLayout="column" fxLayoutGap="2%" fxLayoutAlign="flex-start">
    <ng-container *ngFor="let household of households">
      <div fxFlex>
        <mat-card>
          <div fxLayout="column" fxLayoutGap="1%">
            <mat-card-title>{{ household.name }}</mat-card-title>
            <mat-divider></mat-divider>
            <mat-card-content fxLayout="column">
              <mat-expansion-panel>
                <mat-expansion-panel-header>
                  <mat-panel-title>
                    <mat-card-subtitle>{{ household.symbol }}</mat-card-subtitle>
                  </mat-panel-title>
                </mat-expansion-panel-header>
                <div fxLayout="column">
                  <p>{{ household.description }}</p>
                  <p>{{ household.street }}</p>
                  <p>{{ household.city }}</p>
                  <p>{{ household.country }}</p>
                </div>
              </mat-expansion-panel>
            </mat-card-content>
            <mat-card-actions>
              <button mat-button>EDIT</button>
              <button mat-button>REMOVE</button>
            </mat-card-actions>
          </div>
        </mat-card>
      </div>
    </ng-container>
    <button mat-stroked-button color="primary" (click)="onLoadMore()">Load More</button>
  </div>
</div>
