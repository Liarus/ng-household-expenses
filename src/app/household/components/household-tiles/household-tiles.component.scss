.tile-nowrap {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.mat-card .mat-divider-horizontal {
  position: relative;
}

.mat-expansion-panel-header {
  padding: 0;
  padding-right: 0.5em;
}

.mat-expansion-panel {
  box-shadow: none;
}

::ng-deep .mat-expansion-panel-body {
    padding: 0 !important;
}

::ng-deep .cdk-overlay-backdrop {
  pointer-events: none;
}

.mat-card-subtitle, .mat-card-actions {
  margin-bottom: 0;
}

.mat-card-actions {
  margin-left: 0px;
  margin-right:08px;
  padding: 8px 0;
}
