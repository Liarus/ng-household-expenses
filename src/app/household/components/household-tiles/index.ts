export * from './household-tiles.component';
