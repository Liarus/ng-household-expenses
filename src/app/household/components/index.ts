export * from './household-list';
export * from './household-dialog';
export * from './household-tiles';
export * from './household-search';
