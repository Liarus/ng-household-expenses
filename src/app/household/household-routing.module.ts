import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';

import { HouseholdPageComponent } from './containers';

const routes: Routes = [{ path: '', component: HouseholdPageComponent }];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class HouseholdRoutingModule {}
