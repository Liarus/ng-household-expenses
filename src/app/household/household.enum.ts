export enum HouseholdSorting {
  NameAsc = 1,
  NameDesc = 2,
  SymbolAsc = 3,
  SymbolDesc = 4,
  DescriptionAsc = 5,
  DescriptionDesc = 6
}
