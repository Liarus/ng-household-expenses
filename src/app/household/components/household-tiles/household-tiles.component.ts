import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter } from '@angular/core';

import { Household } from '../../models/household.model';
import { HouseholdFilter } from '../../models/householdFilter.model';

@Component({
  selector: 'app-household-tiles',
  changeDetection: ChangeDetectionStrategy.OnPush,
  templateUrl: './household-tiles.component.html',
  styleUrls: ['./household-tiles.component.scss']
})
export class HouseholdTilesComponent {
  @Input() households: Household[];
  @Input() isLoading: boolean;
  @Input() itemCount: number;

  @Output() filterChanged = new EventEmitter<Partial<HouseholdFilter>>();
  @Output() edit = new EventEmitter<string>();
  @Output() remove = new EventEmitter<string>();

  private pageIndex = 1;

  onLoadMore() {
    if (this.households && this.households.length !== this.itemCount) {
      this.pageIndex ++;
      this.filterChanged.emit({
        pageNumber: this.pageIndex
      });
    }
  }
}
