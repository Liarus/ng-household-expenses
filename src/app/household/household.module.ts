import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { HouseholdEffects } from './store/effects/household.effects';
import { reducers } from './store/reducers';
import { HouseholdRoutingModule } from './household-routing.module';
import { MaterialModule } from '../material/material.module';
import { HouseholdPageComponent } from './containers';
import { HouseholdListComponent, HouseholdDialogComponent } from './components';
import { FlexLayoutModule } from '@angular/flex-layout';

export const COMPONENTS = [
    HouseholdPageComponent,
    HouseholdListComponent,
    HouseholdDialogComponent
];

@NgModule({
  imports: [
    CommonModule,
    ReactiveFormsModule,
    HouseholdRoutingModule,
    MaterialModule,
    FlexLayoutModule,
    StoreModule.forFeature('households', reducers),
    EffectsModule.forFeature([HouseholdEffects]),
  ],
  declarations: COMPONENTS,
  entryComponents: [
    HouseholdDialogComponent
  ]
})
export class HouseholdModule {}
