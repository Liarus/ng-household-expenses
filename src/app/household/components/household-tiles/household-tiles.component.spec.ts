import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

import { HouseholdTilesComponent } from './household-tiles.component';
import { Household } from '../../models/household.model';
import { MaterialModule } from '../../../material/material.module';
import { TEST_DATA } from '../../../shared/tests/test-data';

describe('HouseholdTilesComponent', () => {
  let component: HouseholdTilesComponent;
  let fixture: ComponentFixture<HouseholdTilesComponent>;

  const households = TEST_DATA.household.households as Household[];

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        MaterialModule,
        NoopAnimationsModule
      ],
      declarations: [ HouseholdTilesComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HouseholdTilesComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot when with households', () => {
    component.isLoading = false;
    component.households = households;
    component.itemCount = households.length;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  describe('onLoadMore', () => {
    it('should chenge filter when more households exist', () => {
      component.isLoading = false;
      component.households = households;
      component.itemCount = households.length - 1;
      spyOn(component.filterChanged, 'emit');

      fixture.detectChanges();
      component.onLoadMore();

      expect(component.filterChanged.emit).toHaveBeenCalled();
    });

    it('should not change filter if all households rendered', () => {
      component.isLoading = false;
      component.households = households;
      component.itemCount = households.length;
      spyOn(component.filterChanged, 'emit');

      fixture.detectChanges();
      component.onLoadMore();

      expect(component.filterChanged.emit).not.toHaveBeenCalled();
    });
  });
});
