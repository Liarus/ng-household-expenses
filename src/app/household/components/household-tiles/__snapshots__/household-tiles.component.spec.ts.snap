// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseholdTilesComponent should match snapshot when with households 1`] = `
<app-household-tiles
  edit={[Function EventEmitter]}
  filterChanged={[Function EventEmitter]}
  households={[Function Array]}
  isLoading="false"
  itemCount={[Function Number]}
  pageIndex={[Function Number]}
  remove={[Function EventEmitter]}
>
  <div
    fxlayout="column"
  >
    
    <div
      fxlayout="column"
      fxlayoutalign="flex-start"
      fxlayoutgap="1em"
    >
      
      <button
        class="mat-stroked-button mat-primary _mat-animation-noopable"
        color="primary"
        mat-stroked-button=""
        ng-reflect-color="primary"
      >
        <span
          class="mat-button-wrapper"
        >
          Load More
        </span>
        <div
          class="mat-button-ripple mat-ripple"
          matripple=""
          ng-reflect-centered="false"
          ng-reflect-disabled="false"
          ng-reflect-trigger="[object HTMLButtonElement]"
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
    </div>
  </div>
</app-household-tiles>
`;
